### PR TITLE
{tools}[GCCcore/12.3.0] re2c v3.1

### PR DESCRIPTION
**Pull request by maxim-masterov**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/r/re2c/re2c-3.1-GCCcore-12.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+traditional table-driven approach, re2c encodes the generated finite state automata directly in the form
+of conditional jumps and comparisons."""
+
+toolchain = {'name': 'GCCcore', 'version': '12.3.0'}
+
+source_urls = ['https://github.com/skvadrik/re2c/releases/download/%(version)s']
+sources = ['%(name)s-%(version)s.tar.xz']
+checksums = ['0ac299ad359e3f512b06a99397d025cfff81d3be34464ded0656f8a96676c029']
+
+builddependencies = [('binutils', '2.40')]
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/r/re2c/re2c-3.1-GCCcore-12.3.0.eb:18]: On generoso the build failed with:
```
configure: error: no suitable Python interpreter found
```
```suggestion
builddependencies = [
    ('binutils', '2.40'),
    ('Python', '3.11.3')
]
```

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
zen2-rockylinux-88 - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7452 32-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/SebastianAchilles/d92ad2bef641283bf4ff743a3079aca9 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bwd-rockylinux-92 - Linux Rocky Linux 9.2 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), 2 x NVIDIA NVIDIA GeForce GTX 1060 6GB, 535.54.03, Python 3.9.16
See https://gist.github.com/SebastianAchilles/b099e11deadaca94b9a30071932bfa64 for a full test report.

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen2

File: easybuild/easyconfigs/r/re2c/re2c-3.1-GCCcore-12.3.0.eb (hunk 1 above)
Comment (maxim-masterov) [reply easybuild/easyconfigs/r/re2c/re2c-3.1-GCCcore-12.3.0.eb]: thanks for fixing

**Review by SebastianAchilles — APPROVED** [review]
lgtm

**Comment by SebastianAchilles** [comment]
Going in, thanks @maxim-masterov!

**Pull request merged**